crates/iota/src/client_commands.rs

```diff
@@ -3047,3 +3051,33 @@ fn parse_emit_option(s: &str) -> Result<HashSet<EmitOption>, String> {
 
     Ok(options)
 }
+
+/// Warn the user if the CLI falls behind the current on-chain protocol
+/// version.
+async fn check_protocol_version_and_warn(client: &IotaClient) -> Result<(), anyhow::Error> {
+    let current_protocol_version = client
+        .read_api()
+        .get_protocol_config(None)
+        .await?
+        .protocol_version
+        .as_u64();
+
+    let build_protocol_version = ProtocolVersion::MAX.as_u64();
+    if build_protocol_version != current_protocol_version {
```

Comment:
This would also create a warning if the CLI version is newer than the protocol version, don't we want to use `<` instead? Because if a new version is published, it will take some time before the nodes upgrade to it

Reply:
Another node won't help as all nodes will upgrade to the new protocol version at the same time
Just the error message is a bit wrong in the case when the cli version is newer, as in this case one probably has the latest version installed already

Reply:
But your comment has `Consider`, I guess it's fine this way, we can still change later if it will get annoying

Reply:
I agree that the warning is not accurate for this behavior. and wonder if it's worth to distinguish between the two scenarios that the CLI is too old or too new. How  likely is it though that someone uses the CLI before the network has been upgraded? And do you not think that a version mismatch is always something to bring to the user's attention? If I'ld handle the "too new" scenario, then we could have a message saying, that the network as not yet been upgraded, so the user can decide to risk doing the operation, or wait. Wdyt?

Reply:
Changed the message depending on the mismatch scenario: https://github.com/iotaledger/iota/pull/5042/commits/f5a8675410bf2dc474d34bbd4d52c4a25231c27d